src/components/transactions/StakeRewardClaimRestake/StakeRewardClaimRestakeModalContent.tsx

```diff
@@ -0,0 +1,149 @@
+import { TxErrorView } from '../FlowCommons/Error';
+import { GasEstimationError } from '../FlowCommons/GasEstimationError';
+import { TxSuccessView } from '../FlowCommons/Success';
+import { DetailsNumberLine, TxModalDetails } from '../FlowCommons/TxModalDetails';
+import { TxModalTitle } from '../FlowCommons/TxModalTitle';
+import { ChangeNetworkWarning } from '../Warnings/ChangeNetworkWarning';
+import { StakeRewardClaimRestakeActions } from './StakeRewardClaimRestakeActions';
+
+export type StakeRewardClaimRestakeProps = {
+  stakeAssetName: string;
+  icon: string;
+};
+
+export enum ErrorType {
+  NOT_ENOUGH_BALANCE,
+}
+
+type StakingType = 'aave' | 'bpt';
+
+export const StakeRewardClaimRestakeModalContent = ({
+  stakeAssetName,
+  icon,
+}: StakeRewardClaimRestakeProps) => {
+  const { chainId: connectedChainId, readOnlyModeAddress } = useWeb3Context();
+  const { gasLimit, mainTxState: txState, txError } = useModalContext();
+  const { currentNetworkConfig, currentChainId } = useProtocolDataContext();
+
+  const { data: stakeUserResult } = useUserStakeUiData();
+  const { data: stakeGeneralResult } = useGeneralStakeUiData();
+  const stakeData = stakeGeneralResult?.[stakeAssetName as StakingType];
+
+  // hardcoded as all rewards will be in aave token
+  const rewardsSymbol = 'AAVE';
+  const [_amount, setAmount] = useState('');
+  const amountRef = useRef<string>();
+
+  const maxAmountToClaim = normalize(
+    stakeUserResult?.[stakeAssetName as StakingType].userIncentivesToClaim || '0',
+    18
+  );
+  const isMaxSelected = _amount === '-1';
+  const amount = isMaxSelected ? maxAmountToClaim : _amount;
+
+  const handleChange = (value: string) => {
+    const maxSelected = value === '-1';
+    amountRef.current = maxSelected ? maxAmountToClaim : value;
+    setAmount(value);
+  };
+
+  // staking token usd value
+  const amountInUsd =
+    Number(maxAmountToClaim) *
+    (Number(normalize(stakeData?.stakeTokenPriceEth || 1, 18)) *
+      Number(normalize(stakeGeneralResult?.ethPriceUsd || 1, 8)));
+
+  // error handler
+  let blockingError: ErrorType | undefined = undefined;
+  if (maxAmountToClaim === '0') {
+    blockingError = ErrorType.NOT_ENOUGH_BALANCE;
+  }
+
+  const handleBlocked = () => {
+    switch (blockingError) {
+      case ErrorType.NOT_ENOUGH_BALANCE:
+        return <Trans>No rewards to claim</Trans>;
+      default:
+        return null;
+    }
+  };
+
+  // is Network mismatched
+  const stakingChain =
+    currentNetworkConfig.isFork && currentNetworkConfig.underlyingChainId === stakeConfig.chainId
+      ? currentChainId
+      : stakeConfig.chainId;
+  const isWrongNetwork = connectedChainId !== stakingChain;
+
+  const networkConfig = getNetworkConfig(stakingChain);
+
+  if (txError && txError.blocking) {
+    return <TxErrorView txError={txError} />;
+  }
+  if (txState.success)
+    return (
+      <TxSuccessView action={<Trans>Restaked</Trans>} amount={amount} symbol={rewardsSymbol} />
```

Comment:
For the amount prop here, pass in `amountRef.current`. The `amount` value will change when we refetch the stake data after a successful tx, so it won't show the correct amount that was claimed.

This also needs to be updated in the StakerewardClaimModal component

Reply:
I'm wondering why we're changing this now. As previously was amount? 

StakerewardClaimModal shouldn't have been worked on.

Reply:
Must have been an existing bug in that modal